godplugin/module_config.go

```diff
@@ -0,0 +1,55 @@
+package godplugin
+
+import (
+	"os"
+
+	"gopkg.in/yaml.v2"
+)
+
+var modConfigDefaults = func() rawConfig {
+	return rawConfig{
+		"update_every":        1,
+		"autodetection_retry": 0,
+	}
+}
+
+type rawConfig map[string]interface{}
+
+func (r *rawConfig) merge(src rawConfig) {
+	for key, val := range src {
+		if _, ok := (*r)[key]; !ok {
+			(*r)[key] = val
+		}
+	}
+}
+
+type moduleConfig struct {
+	Global rawConfig   `yaml:"global"`
+	Jobs   []rawConfig `yaml:"jobs"`
+}
+
+func (m *moduleConfig) load(filename string) error {
+	file, err := os.Open(filename)
+	defer file.Close()
```

Comment:
move to below err check

Reply:
reason?

Reply:
if err is not nil, file may be nil

Reply:
`Close` can handle it, since we are not checking `Close` result it's OK to leave it here